app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,81 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def tasks_complete
+    @tasks = Task.all
+  end
+
+  def new
+      @task = Task.new
+  end
+
+  def create
+    task = Task.new(task_params)
+
+
+    if task.save
+      redirect_to tasks_path
+    end
+  end
+
+  def show
+    task_id = params[:id]
+
+   @task = Task.find(task_id)
+
+  end
+
+
+  def edit
+    @task = Task.find(params[:id])
+  end
+
+  def update
+    task = Task.find(params[:id])
+
+    task.assign_attributes(task_params)
+
+    if task.save
+      redirect_to task_path(task)
+    end
+  end
+
+  def complete
+    @task = Task.find(params[:id])
+    @task.completion_date = Date.today
+    @task.status = "closed"
+    @task.priority = nil
+    @task.save
+
+    if @task.save
+      redirect_to tasks_path
+    end
+  end
+
+  def uncomplete
```

Comment:
I think you could consolidate these two methods into one - you could have one method that would just toggle the values instead of having the UI figure out which one to use